laurent22/joplin
# Desktop,Mobile: Resolves #10031: Upgrade CodeMirror 6 packages

# Summary

Upgrades CodeMirror 6 packages, resolving #10031.

# Notes

This pull request **does not** update `@codemirror/language-data` -- doing so increases the mobile bundle size from 1.7 MB to 2.2 MB. Additionally, it seems to trigger Webpack's chuncking mechanism (though we can disable it with `output.chunkFormat = false`).

# Testing

1. Create a new document with the following content:
````markdown
# Test

>
> This is a block quote.
>

Should have link formatting:
[This is a link](https://joplinapp.org/). https://this.is.a.test.joplinapp.org/

Should auto-renumber:
1. This
2. is
    1. This
    2. is
    3. a
    4. test
5. a
6. list
7. testing...


`shouldnotspellcheck` Should spllchek.

```javascript
function foo() {
    // This should be highlighted.
}
```

# Test 2
## Test 3
### Test 4

```yaml
a:
    b:
    - c
    - d
```

| a | b | c|
|---|---|--|
|  |   |   |
|test| | |
````
2. Verify that links, code blocks, headings, and block quotes have the correct styling in the beta editor.
3. Verify that the table is in a monospaced font.
4. Verify that numbered list continuation works.
5. Select everything
    - Verify that "select all" works on iOS
6. Duplicate it
7. Repeat steps 5 and 6 three more times.
8. (Desktop only) Enable vim keyboard emulation
9. Jump to the top of the document (`gg`) then to the end (`G`)
10. Go to the middle of the numbered list
11. Open a new line (`o`)

This has been tested successfully on:
- [x] Ubuntu 23.10
- [x] Android 13

**Note**: On iOS, the inline code region is spellchecked (despite having `spellcheck="false"`), but this doesn't seem to be a regression (comparing with latest beta).
